dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/weld/BeanDeploymentArchiveImpl.java

```diff
@@ -476,11 +480,13 @@ private void scanForEndpoints() throws CDIException {
                     for (EjbDescriptor<?> ejbDescriptor : ejbDescriptors) {
                         boolean added = false;
                         for (WebSphereBeanDeploymentArchive child : getWebSphereBeanDeploymentArchives()) {
-                            if ((child.getAllClazzes().containsKey(ejbDescriptor.getBeanClass().getName()))
-                                && (child.getAllClazzes().containsValue(ejbDescriptor.getBeanClass()))) {
-                                child.addEjbDescriptor(ejbDescriptor);
-                                added = true;
-                                break;
+                            if (child.getAllClazzes().contains(ejbDescriptor.getBeanClass().getName())) {
```

Comment:
Is it safe to only check by name? I think you need to check that the classloader matches too?